engine/base.ftl

```diff
@@ -745,7 +746,9 @@ are added.
                         "Values" : asArray(attribute.Values![]),
                         "Children" : asArray(attribute.Children![]),
                         "SubObjects" : attribute.SubObjects!attribute.Subobjects!false,
-                        "PopulateMissingChildren" : attribute.PopulateMissingChildren!true
+                        "PopulateMissingChildren" : attribute.PopulateMissingChildren!true,
+                        "AttributeSet" : attribute.AttributeSet!"",
+                        "Component" : attribute.Component!""
```

Comment:
There is no mention of component in the PR description. I understand what an attributeSet is doing, but what is component doing?

Reply:
Component defines a SubComponent within the composite object model. It is not intended for further use, or manual assignment within `id.ftl` files, rather it separates a reference to another component schema out from a reference to an AttributeSet.

A method to differentiate the two is necessary to provide accurate schema references, that define both the "SchemaType" (component, reference, attributeset) and the Instance (baseline, alertprofile, link, etc) values.

Reply:
I've updated the PR description to include this.